src/cc/bcc_elf.c

```diff
@@ -122,10 +128,26 @@ static int listprobes(Elf *e, bcc_elf_probecb callback, const char *binpath,
   Elf_Scn *section = NULL;
   size_t stridx;
   int elf_class = gelf_getclass(e);
+  uint64_t first_inst_offset = 0;
 
   if (elf_getshdrstrndx(e, &stridx) != 0)
     return -1;
 
+  // Get the offset to the first instruction
+  while ((section = elf_nextscn(e, section)) != 0) {
```

Comment:
Nit: Maybe you can update `bcc_elf_foreach_load_section` to be more general, so that it could take different filtering condition (note section rather than load section) and reuse the same code.
Up to your call:))

Reply:
I will leave it now. We have a couple of other while loops for sections as well. Maybe somebody else can volunteer to do this in the future?